lib/plugins/inspec-init/lib/inspec-init/cli_profile.rb

```diff
@@ -24,14 +24,24 @@ def valid_profile_platforms
              desc: "Which platform to generate a profile for: choose from #{valid_profile_platforms.join(", ")}"
       option :overwrite, type: :boolean, default: false,
              desc: "Overwrites existing directory"
+      option :resource_dependency, default: nil, type: :string, aliases: [:r],
+              desc: "Generate a resource dependency with the profile"
       def profile(new_profile_name)
         Inspec.with_feature("inspec-cli-init-profile") {
           unless valid_profile_platforms.include?(options[:platform])
             ui.error "Unable to generate profile: No template available for platform '#{options[:platform]}' (expected one of: #{valid_profile_platforms.join(", ")})"
             ui.exit(:usage_error)
           end
-          template_path = File.join("profiles", options[:platform])
+          # Get the index of "--resource-dependency" in ARGV
+          resource_index = ARGV.index("--resource-dependency")
 
+          # Check if "--resource-dependency" exists but has no argument
+          if resource_index && (resource_index + 1 == ARGV.size || ARGV[resource_index + 1].start_with?("--"))
```

Comment:
Instead of using ARGV, can you instead try taking the approach of using `options[:resource_dependency]` for `nil` handling?

Reply:
Also check, that even though this new key has a default value of `nil` it seems to be returning `"resource_dependency"=>"resource_dependency"` in thor options when nothing is passed in resource dependency. Please have a check on this behaviour.